src/util/parallel/dune

```diff
@@ -0,0 +1,18 @@
+(include_subdirs no)
+
+(library
+ (name goblint_parallel)
+ (public_name goblint.parallel)
+ (libraries
+   batteries
+   (select gobMutex.ml from
+      (domainslib -> gobMutex.domainslib.ml)
+      ( -> gobMutex.no-domainslib.ml)
+   )
+   (select threadpool.ml from 
+      (domainslib -> threadpool.domainslib.ml)
+      (-> threadpool.no-domainslib.ml)
```

Comment:
I think it's a good idea to have `.mli` files for these as well (not dependent on domainlib presence).

Reply:
Should be introduced now.